### PR TITLE
DrivAerML: metric-aware w=0.05 + T_max=20 (shorter schedule)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
Test whether a shorter cosine schedule (T_max=20 vs champion T_max=30) with metric-aware loss finds a different optimum. Shorter cycles mean more frequent restarts. T_max=20 was inferior on MSE-only (4.943%), but metric-aware loss may change the landscape.

## Instructions
```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml --dm-loss mse_plus_raw_rel_l2 --dm-rel-l2-weight 0.05 \
  --optimizer adamw --lr 5e-4 --cosine-t-max 20 --grad-clip 0.5 \
  --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 \
  --ema-decay 0.9995 --ema-mode fixed --no-compile-model --save-checkpoint \
  --eval-interval 10 --seed 42 --wandb-group dm-metric-aware-schedule \
  --kill-if-best-val-above 150:8.0 --kill-if-no-improvement 200:0.05
```
**Target:** val < 3.700%

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT mitsuha:

## Results

### DrivAerML: metric-aware w=0.05 + T_max=20 (shorter cosine schedule)

| Metric | Baseline (PR #3302, T_max=30) | This experiment (T_max=20) |
|--------|-------------------------------|----------------------------|
| val_primary/surface_rel_l2_pct | **3.700%** (epoch 780) | **4.019%** (epoch 970) |
| test_primary/surface_rel_l2_pct | — | **4.610%** |

### Key metrics
- **Best val:** 4.019% at epoch 970 (target was < 3.700%)
- **Test:** 4.610% (from best checkpoint at epoch 970)
- **Final val (epoch 999):** 4.044%
- **Training loss (final):** 0.0035
- **Peak GPU memory:** ~7.4 GB
- **Total epochs:** 999 (hit max epoch limit)

### Command used
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml --dm-loss mse_plus_raw_rel_l2 --dm-rel-l2-weight 0.05 \
  --optimizer adamw --lr 5e-4 --cosine-t-max 20 --grad-clip 0.5 \
  --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 \
  --ema-decay 0.9995 --ema-mode fixed --no-compile-model --save-checkpoint \
  --eval-interval 10 --seed 42 --wandb-group dm-metric-aware-schedule \
  --kill-if-best-val-above 150:8.0 --kill-if-no-improvement 200:0.05 \
  --agent mitsuha --wandb-name "mitsuha/dm-metric-aware-w005-tmax20"
```

### W&B run
- Run ID: `r6bo7elk`
- URL: https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/r6bo7elk

### What happened

**Negative result.** T_max=20 with metric-aware loss converges to ~4.02% val, which is **0.32pp worse** than the T_max=30 baseline (3.70%). The hypothesis that shorter cosine cycles might find a different optimum under metric-aware loss did not hold.

Convergence trajectory:
- Rapid early convergence: 41% → 13% in first 50 epochs
- Plateaued around 4.2% by epoch 400-500
- Very slow continued descent: 4.2% → 4.02% over epochs 500-999
- Improvement rate was ~0.02pp per 30 epochs in the 600-999 range

The pattern is consistent with previous MSE-only results where T_max=20 was also inferior (4.943% vs T_max=30). Metric-aware loss narrows the gap somewhat (from ~1.24pp to ~0.32pp difference), suggesting the metric-aware loss helps T_max=20 more than T_max=30, but T_max=30 remains clearly superior.

Training was stable throughout — no NaN, no divergence, grad clipping events decreased to near-zero after ~300 epochs (with occasional spikes at cycle boundaries).

### Suggested follow-ups
- **T_max=25 or T_max=35**: The optimal cosine period for metric-aware loss might differ from MSE-only — worth testing adjacent values
- **Warm restarts with decay**: Instead of fixed T_max, try increasing T_max each restart (e.g., T_max doubles) to combine broad exploration early with fine-tuning later
- **T_max=30 with different seeds**: The 3.70% baseline was seed=42; confirming robustness across seeds may be more valuable than schedule tuning

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3332 because ADVISOR: Closing — val=4.019% with T_max=20 metric-aware. T_max=20 is consistently worse across all configs. T_max=36 is the clear optimum..
